STYLEGUIDE.md

```diff
@@ -0,0 +1,716 @@
+
+There should be no empty lines in between, except if it is part of the description.
+
+### Use `@deprecated` tag for deprecation
+When deprecating public API, this is done through the TypeDoc comments.
+The first line of the comment should start with `@deprecated` followed by a note mentioning since which version this is deprecated (e. g. `Since v1.0.0.`) and a note of what to use instead (or alternatively that it won't be replaced).
+
+❌ Examples of **incorrect** code:
+```ts
+/* Don't leave out from which version on this is deprecated */
+/**
+ * @deprecated Use [[foo]] instead.
+ */
+
+/* Don't leave out what to use instead */
+/**
+ * @deprecated Since v1.2.3.
+ */
+```
+
+✅ Examples of **correct** code:
+```ts
+/* Add all deprecation information */
+/**
+ * @deprecated Since v1.2.3. Use [[foo]] instead.
+ */
+```
+
+### Don't reference types in parameters an type parameters
+In JSDoc it is common to specify the types of parameters and type parameters as JavaScript has no types.
+In TypeDoc this is not necessary, because TypeScript inherently has types and those should not differ between the implementation and documentation.
+A parameter or type parameter line should start with `@param` or `@typeparam`, followed by the (type) parameter name, a dash, and the description.
+The description should end with a full stop.
+
+❌ Examples of **incorrect** code:
+```ts
+/* Don't use types for the parameters */
+/**
+ * @param param {string} - The description.
+ */
+
+/* Don't skip the dash */
+/**
+ * @param param The description.
+ */
+
+/* Don't leave out the fullstop */
+/**
+ * @param param - The description
+ */
+```
+
+✅ Examples of **correct** code:
+```ts
+/* Apply all rules from above */
+/**
+ * @param param - The description.
+ */
+```
+
+### Use @returns if a function has a return value ✓
+Functions that return something should have an `@returns` statement in the documentation, followed by a description of the return value.
+The description should end with a full stop.
+Similar to parameters, return types should not be referenced in the documentation.
+For consistency, `@return` should not be used.
+The only exceptions are asynchronous functions that return `Promise<void>`.
+They technically return a promise, which does not need an extra description.
+
+❌ Examples of **incorrect** code:
+```ts
+/* Don't use types in the documentation */
+/**
+ * @returns {string} The description of the return value.
+ */
+
+/* Don't use @return */
+/**
+ * @return The description of the return value.
+ */
+
+/* Don't leave out the fullstop */
+/**
+ * @returns The description of the return value
+ */
+```
+
+✅ Examples of **correct** code:
+```ts
+/* Apply all rules from above */
+/**
+ * @returns The description of the return value.
+ */
+```
+
+
+## Tests
+
+### Use .spec in test file names
+Use `.spec` not `.test` to differentiate test files.
+
+❌ Examples of **incorrect** code:
+```ts
+/* Don't use .test.ts */
+test-file.test.ts
```

Comment:
```suggestion
/* Don't use .test.ts */
test-file.test.ts

/* Don't put unit tests in a separate folder */
test/file.ts
```

Reply:
I realized later that this is a duplicate for the next point in the style guide but I think it's best to include here as people may not read the guide front-to-back.

Reply:
Meh, maybe if we move it around?

Reply:
I will resolve this for now, as I think the TOC should help here. I agree that things might be difficult to find, so maybe we can find better categories, but I wouldn't put different aspects into one rule.